File: GroceryShoppingListApp/src/GroceryShoppingListApp.Core/Model/ListAggregate/GroceryList.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace GroceryShoppingListApp.Core;

public class GroceryList
{
    public Guid GroceryListId { get; set; }
    public Guid UserId { get; set; }

    /// <summary>
    /// Gets or sets the tenant ID for multi-tenant isolation.
    /// </summary>
    public Guid TenantId { get; set; }
    public string Name { get; set; } = string.Empty;
    public DateTime CreatedDate { get; set; }
    public bool IsCompleted { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public ICollection<GroceryItem> Items { get; set; } = new List<GroceryItem>();
}

File: GroceryShoppingListApp/src/GroceryShoppingListApp.WebApp/projects/grocery-shopping-list-app/src/app/models/grocery-list.ts
export interface GroceryList {
  groceryListId: string;
  userId: string;
  name: string;
  createdDate: Date;
  isCompleted: boolean;
  createdAt: Date;
}

export interface CreateGroceryListRequest {
  userId: string;
  name: string;
}

export interface UpdateGroceryListRequest {
  groceryListId: string;
  name: string;
  isCompleted: boolean;
}
